docs/nym-simple-websocket-tutorial.md

```diff
@@ -0,0 +1,1212 @@
+    ```
+
+    We're just looking to make sure we have our typescript file working when we run the application in the browser. We'll check this is just a moment.
+    
+    Once we have done that , navigate into our `package.json` and in the "scripts" section, just above the test command,  paste the following:
+
+    ```
+    "start": "parcel src/index.html"
+    ```
+
+    Now back in our terminal , type `npm start` and press enter.
+
+    <img src="/img/tutorials/simple-websocket/image4.png"/>
+
+    This will result in the above output. Open your browser to see the results [(localhost:1234)](http://localhost:1234/). You should now have a running web application with 'Test' printed on the browser window!
+
+    The `console.log` statement in the above code can be checked by right clicking ⇧ on the browser and selecting 'Inspect' and navigating to the 'Console' section of the pane that appears. You should see 'test log' printed there. 
+
+    
+### Building the User Client
+
+We will focus mainly on the logic of the `index.ts` file in this section. It is where we want to establish our logic which connects our application to the mixnet, construct messages that we want to send and then send it to the mixnet for a service provider to receive.
+
+ Remember that our aim is to get two Typescript applications running in the first [diagram we saw](#what-are-we-building) (the blue boxes left and right of the image). Everything in the middle (the Mixnet aka the green box in the diagram) has already been taken care of for us. The first place to to start will be implementing the code that connects our Typescript Client to our Nym Websocket Client (the orange box to the right of our 'User Client').
+
+ 1. In our `index.ts`, remove the existing `console.log` statement an paste or type in the following code : 
+    
+    ```
+    async function main() {
+    }
+
+    /* Connect to a websocket. */
+    function connectWebsocket(url) {
+        return new Promise(function (resolve, reject) {
+            var server = new WebSocket(url);
+            console.log('connecting to Websocket Server (Nym Client)...')
+            server.onopen = function () {
+                resolve(server);
+            };
+            server.onerror = function (err) {
+                reject(err);
+            };
+        
+        });
+    }
+    
+    main();
+    ```
+    * `main()` - This function will be the first function that we will want to declare our file. It is where the main bulk of logic is 'queued up to run' within the application. Think of the function as the 'function in charge' that lists what order we should run our logic in to make our Application work as desired for the user. This is exactly where we will want to try to make a connection with the Nym Websocket Client. We also want to call this at the bottom, so we can et our code know that it has to be executed on starting up the Application when we open it in the browser (eventually).
```

Comment:
First sentence is missing an 'in': 'this function will be the first function that will we want to declare _in_ our file'?

Reply:
Also we can strip this down a little I think. I don't think we need to be explaining what a `main()` function does in too much detail